src/LndClient/Data/Invoice.hs

```diff
@@ -33,6 +44,39 @@ instance FromGrpc Invoice GRPC.Invoice where
       <*> fromGrpc (GRPC.invoicePaymentRequest x)
       <*> fromGrpc (GRPC.invoicePrivate x)
       <*> fromGrpc (GRPC.invoiceAddIndex x)
-      <*> first
-        (\e -> FromGrpcError $ "Invalid Invoice State" <> show e)
-        (enumerated $ GRPC.invoiceState x)
+      <*> ( fromGrpc
+              =<< first
+                (\e -> FromGrpcError $ "Invalid Invoice State" <> show e)
+                (enumerated $ GRPC.invoiceState x)
+          )
+
+instance C2.FromGrpc Invoice LnGRPC.Invoice where
+  fromGrpc x =
+    Invoice
+      <$> fromGrpc (x ^. LnGRPC.rHash)
+      <*> fromGrpc (x ^. LnGRPC.amtPaidMsat)
+      <*> fromGrpc (x ^. LnGRPC.valueMsat)
+      <*> fromGrpc (x ^. LnGRPC.settled)
+      <*> fromGrpc (x ^. LnGRPC.settleIndex)
+      <*> fromGrpc (x ^. LnGRPC.memo)
+      <*> fromGrpc (fromStrict $ x ^. LnGRPC.paymentRequest)
+      <*> fromGrpc (x ^. LnGRPC.private)
+      <*> fromGrpc (x ^. LnGRPC.addIndex)
+      <*> C2.fromGrpc (x ^. LnGRPC.state)
+
+instance C2.FromGrpc InvoiceState LnGRPC.Invoice'InvoiceState where
+  fromGrpc x =
+    case x of
+      LnGRPC.Invoice'OPEN -> Right OPEN
+      LnGRPC.Invoice'SETTLED -> Right SETTLED
+      LnGRPC.Invoice'CANCELED -> Right CANCELED
+      LnGRPC.Invoice'ACCEPTED -> Right ACCEPTED
+      _ -> Left $ FromGrpcError "InvoiceState value cannot be recognized"
+
+instance FromGrpc InvoiceState GRPC.Invoice_InvoiceState where
+  fromGrpc x =
+    case x of
```

Comment:
right before case

Reply:
done

Reply:
there is one left at the bottom, so right cannot be out